pkg/plugins/k8s/k8s_plugin.go

```diff
@@ -358,164 +219,101 @@ func (p *K8sPlugin) sriovServicesStateUpdate() error {
 		}
 		// create and enable the service if it doesn't exist or is not enabled
 		if !isServiceEnabled {
-			*s.update = true
+			s.update.SetNeedReboot()
 		} else {
-			*s.update = p.isSystemServiceNeedUpdate(s.srv)
-		}
-		if *s.update {
-			p.updateTarget.systemServices = append(p.updateTarget.systemServices, s.srv)
+			if p.isSystemDServiceNeedUpdate(s.srv) {
+				s.update.SetNeedReboot()
+			}
 		}
 	}
-
 	return nil
 }
 
-func (p *K8sPlugin) getSwitchDevSystemServices() []*hostTypes.Service {
-	return []*hostTypes.Service{p.networkManagerService, p.openVSwitchService}
-}
-
-func (p *K8sPlugin) isSwitchdevScriptNeedUpdate(scriptObj *hostTypes.ScriptManifestFile) (needUpdate bool, err error) {
-	data, err := os.ReadFile(path.Join(consts.Host, scriptObj.Path))
-	if err != nil {
-		if !os.IsNotExist(err) {
-			return false, err
+func (p *K8sPlugin) updateSriovServices() error {
+	for _, s := range []struct {
+		srv    *hostTypes.Service
+		update *updateTargetReq
+	}{
+		{srv: p.sriovService, update: &p.updateTarget.sriovScript},
+		{srv: p.sriovPostNetworkService, update: &p.updateTarget.sriovPostNetworkScript},
+	} {
+		if s.update.NeedUpdate() {
+			err := p.hostHelper.EnableService(s.srv)
+			if err != nil {
+				return err
+			}
 		}
-		return true, nil
-	} else if string(data) != scriptObj.Contents.Inline {
-		return true, nil
 	}
-	return false, nil
+	return nil
 }
 
-func (p *K8sPlugin) isSwitchdevServiceNeedUpdate(serviceObj *hostTypes.Service) (needUpdate bool, err error) {
-	swdService, err := p.hostHelper.ReadService(serviceObj.Path)
+func (p *K8sPlugin) ovsServiceStateUpdate() error {
+	exist, err := p.hostHelper.IsServiceExist(p.openVSwitchService.Path)
 	if err != nil {
-		if !os.IsNotExist(err) {
-			return false, err
-		}
-		// service not exists
-		return true, nil
+		return err
+	}
+	if !exist {
+		log.Log.Info("k8s plugin systemServicesStateUpdate(): WARNING! openvswitch system service not found, skip update",
```

Comment:
we usually log warning with Error

e.g
`funcLog.Error(nil, "DiscoverVDPAType(): WARNING: unknown VDPA device type for VF, ignore")`
`log.Log.Error(err, "Warning GetOpenstackData(): error getting PCI address for device" `